<launch>
	<arg name="params_folder" value="base" />

	<!-- joystick setup -->
	<remap from="joystick/xinput" to="sputnik_demo/joystick" />
	<remap from="joystick" to="sputnik_demo/joystick" />
	<param name="~dev" value="/dev/input/js1" />
	<node pkg="isc_joy" type="joystick_xbox360" name="joystick_xbox360" />
	<node respawn="true" pkg="joy" type="joy_node" name="sputnik_joy_input">
		<param name="dev" type="string" value="/dev/input/js1" />
		<param name="deadzone" value="0.10" />
	</node>

	<!-- demo setup -->
	<arg name="enable_wandering" value="true" />
	<arg name="enable_route" value="false" />

	<node pkg="sputnik_demo" type="demo" name="sputnik_demo" output="screen"> 
		<param name="enable_wandering" value="$(arg enable_wandering)" />
		<param name="enable_route" value="$(arg enable_route)" />
	</node>

	<!-- waypoint selector -->
	<include file="$(find helper_nodes)/launch/waypoint.launch" if="$(arg enable_route)">
		<arg name="nav_file" value="$(find sputnik_demo)/data/nav.txt" />
	</include>

	<!-- Run the map server -->
	<node name="map_server" pkg="map_server" type="map_server" args="$(find sputnik_nav)/data/maps/$(arg map_filename).pgm 0.05" if="$(arg enable_route)" />

	<!--- Run AMCL -->
	<include file="$(find sputnik_nav)/launch/amcl.launch" if="$(arg enable_route)" />

	<!-- gmapping otherwise -->
	<!-- node pkg="gmapping" type="slam_gmapping" name="mapping_service" unless="$(arg enable_route)" -->
		<!-- param name="map_update_interval" value="2.0" / -->
	<!-- /node -->

	<!-- Run move_base -->
	<remap from="/cmd_vel" to="/RosAria/cmd_vel" />
	<node pkg="move_base" type="move_base" respawn="false" name="move_base" output="log">
		<rosparam file="$(find sputnik_nav)/params/$(arg params_folder)/costmap_common.yaml" command="load" ns="global_costmap" /> 
		<rosparam file="$(find sputnik_nav)/params/$(arg params_folder)/costmap_common.yaml" command="load" ns="local_costmap" />
		<rosparam file="$(find sputnik_nav)/params/$(arg params_folder)/costmap_global.yaml" command="load" />
		<rosparam file="$(find sputnik_nav)/params/$(arg params_folder)/costmap_local.yaml" command="load" /> 
		<rosparam file="$(find sputnik_nav)/params/$(arg params_folder)/planner_local.yaml" command="load" />
	</node>

	<!-- sputnik setup -->
	<include file="$(find sputnik_nav)/launch/sputnik_config.launch">
		<arg name="motor_topic" value="/sputnik_demo/cmd_vel" />
		<arg name="reduced_fov" value="true" />
		<arg name="simulated" value="faLse" />
	</include>
</launch>
